### PR TITLE
feat: prepare manifests for release v0.2.1

### PR DESCRIPTION
**Pull request by iam-veeramalla**
**What type of PR is this?**
> /kind chore

**What does this PR do / why we need it**:
prepare manifests for release v0.2.1

**Have you updated the necessary documentation?**
* [x] Documentation update is required by this PR.
* [x] Documentation has been updated.

**How to test changes / Special notes to the reviewer**:
1. checkout the PR
2. Run the below command to deploy the catalog source 
kubectl apply -f deploy/catalog_source.yaml
3. Install the operator.

### CODE REVIEW

File: build/util/Dockerfile [hunk 1]
```diff
@@ -1,11 +1,11 @@
 # Argo CD v2.2.5
-FROM quay.io/argoproj/argocd@sha256:0cdf5aac80f0a77307cbb288a0c1e5e39d4bac0cafb3c285a6d3c656b9f8e910
+FROM quay.io/argoproj/argocd@sha256:bac1aeee8e78e64d81a633b9f64148274abfa003165544354e2ebf1335b6ee73
 
 USER root
 
 # Ensure system dependencies are installed
 RUN apt-get update && \
-    apt-get install -y curl && \
+    apt-get install -y curl python && \
```
Comment (wtam2018) [inline build/util/Dockerfile:8]: Hmm. I wonder where this `python` dependency come from?

Comment (iam-veeramalla) [reply build/util/Dockerfile:8]: it is removed from argocd recently
https://github.com/argoproj/argo-cd/pull/7947

File: bundle/manifests/argocd-operator.clusterserviceversion.yaml [hunk 2]
```diff
@@ -966,7 +966,7 @@ spec:
                   valueFrom:
                     fieldRef:
                       fieldPath: metadata.annotations['olm.targetNamespaces']
-                image: quay.io/argoprojlabs/argocd-operator:v0.2.0
+                image: quay.io/argoprojlabs/argocd-operator@sha256:a0a4a26e4733613d014f4de193f9596c132fe1b3d8c509a57cd0d1e09b38b611
```
Comment (wtam2018) [inline bundle/manifests/argocd-operator.clusterserviceversion.yaml:969]: do we switch back to a tag (v0.2.1) as some point?

**Review by jopit — CHANGES_REQUESTED** [review]

File: bundle/manifests/argocd-operator.clusterserviceversion.yaml [hunk 3]
```diff
@@ -1048,4 +1048,4 @@ spec:
   provider:
     name: Argo CD Community
   replaces: argocd-operator.v0.1.0
```
[the remark this answers is not in the document]
Comment (iam-veeramalla) [reply bundle/manifests/argocd-operator.clusterserviceversion.yaml]: Thanks for catching @jopit . Made required changes.

File: build/util/Dockerfile (hunk 1 above)
Comment (iam-veeramalla) [reply build/util/Dockerfile]: Thanks @wtam2018 , removed the python package.

**Review by iam-veeramalla — COMMENTED** [review]
LGTM

**Pull request merged**